### PR TITLE
Club members rails app

### PR DESCRIPTION
**Pull request by ammarkandel**
## Rails app to allow users to share their ideas.
- Allow users (create, delete, edit, show) their ideas.
- Allow users to (signup, login, logout).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+This project built to let users share there ideas.
+
+## About
+
+This project is to implement authentication systems so users can only access areas of a site they are authorized to.
+
+## Usage
+
+- Clone the repo "git@github.com:ammarkandel/club-members-rails.git".
+- Open terminal and make sure the path in the project folder and type "rails s".
```
Comment (enaburekhan) [inline README.md]: Add run rails db:create and run rails db:migrate. Before we run rails server

Comment (enaburekhan) [inline README.md]: Sentence in line 12 is incomplete. Kindly rewrite it.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# Club members
+
+This project built to let users share there ideas.
+
+## About
+
+This project is to implement authentication systems so users can only access areas of a site they are authorized to.
```
Comment (enaburekhan) [inline README.md]: Make this about section more descriptive

File: app/controllers/ideas_controller.rb [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  def index
+    @ideas = Idea.all
+  end
+
+  # GET /ideas/1 or /ideas/1.json
+  def show; end
+
+  # GET /ideas/new
+  def new
+    @idea = current_user.ideas.build
```
Comment (enaburekhan) [inline app/controllers/ideas_controller.rb]: This should be: @ idea = Idea.new

**Review by enaburekhan — CHANGES_REQUESTED** [review]
# Status: Required changes:full_moon:
**Hi Ammar!
Great job on this milestone:clap::clap:. The app is working and users can easily sign up and create a post. However, we need to make a few changes to get this project approved:muscle:.**

**Mandatory requirements**

- [x] Make development the default branch in the repository. We should adhere to the correct gitbub workflow.

- [x] As a guest you can see only post's text ( [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/guest_can_see_only_posts_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication)). Currently, as a guest, I also see the author's name. Kindly fix this.

- [ ] it uses `before_action` to restrict access to the `:new` and `:create` methods (uses ApplicationController helper method) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/before_action_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication)). Account for this using both the helper and postController.

[ Optional] 
- Kindly adhere to the rails naming convention. Rails use the conventional names to easily interconnect each file on the backend. let's follow the proper naming conventions(Unless we want to do more work and have a bad time.).

Make the required changes as outline above and also in the PR below. When you are ready, request another review.
Kindly drop your questions/comments here and ping me on slack @ Eric Enaburekhan to respond.
Happy Coding:blush:
Eric Enaburekhan

**Review by MahmoudBakr23 — APPROVED** [review]
# **Approved ✔✔**

<hr>

Hello there, congratulations on getting the requested changes done, and on completing this project, a good thing you used scaffold with the rails generator for making nicely structured files and also for a little bit of styling. **However, I recommend** in future projects try to generate your models and controller files on your own for best practices. Also, in your ```ideas_controller``` file in the ```idea_params``` method, we only permit the needed columns that actually exists within the database table related to that controller in this case (ideas) and it has no column called ```name``` so please remove it to be just ```params.require(:idea).permit(:author_idea)```
Fix the repo's link in the README file (cloning part)
One last thing I recommend, hide the signup and login links when the user visits the signup and/or the login pages.

<hr>

Good luck with the next and as always, happy coding 👨‍💻💻

**Pull request merged**